src/Button/Button.tsx

```diff
@@ -42,64 +52,121 @@ export const Button: FC<Props> = ({
   disabled = false,
   outlined = false,
   handleClick,
-}) => (
-  <Container
-    id={id}
-    className={className}
-    color={color}
-    block={block}
-    inverted={inverted}
-    disabled={disabled}
-    outlined={outlined}
-    onClick={handleClick}
-  >
-    {children}
-  </Container>
-)
-
-const Container = styled.button<IButton>(
-  ({ block, color, inverted, outlined }) => css`
-    position: relative;
-    display: inline-block;
-    box-sizing: border-box;
-    border: none;
-    border-radius: 8px;
-    font-size: 14px;
-    padding: 18px 16px 14px;
-    outline: none;
-    cursor: pointer;
-    width: ${block ? '100%' : 'auto'};
-    background-color: ${theme.colors[`${color}5`]};
-    color: ${theme.colors.white};
+  loading = false,
+  primary = false,
+  secondary = false,
+  tertiary = false,
+  icon = '',
+  forcedWidth = ''
+}) => {
 
-    &:hover:not([disabled]) {
-      background-color: ${theme.colors[`${color}6`]};
-    }
-    &:active:not([disabled]) {
-      background-color: ${theme.colors[`${color}7`]};
-    }
-    &:disabled {
-      opacity: 0.5;
-      cursor: not-allowed;
-    }
-
-    ${(inverted || outlined) &&
-      css`
-        background-color: transparent;
-        border: 1px solid ${outlined ? theme.colors.grey4 : 'transparent'};
-        color: ${outlined ? theme.colors.blue7 : theme.colors[`${color}5`]};
+  return (
+    <div>
+      {
+        primary || secondary || tertiary ? (
+          <Container
+            id={id}
+            disabled={disabled}
+            onClick={handleClick}
+            loading={loading}
+            primary={primary}
+            secondary={secondary}
+            tertiary={tertiary}
+            icon={icon}
+            forcedWidth={forcedWidth}
+          >
+            {icon && (            
+              <IconContainer   
+                render={icon}
+                size={24}
+                color={primary ? 'white' : `${theme.colors.blue7}`} />
+            )}
+            {
+              loading ? (
+                <Loader color={primary ? 'white' : `${theme.colors.blue7}`} height="16" />
+              ) : (
+                children
+            )}
+          </Container>
+        ) : (
+          <LegacyButton
+            id={id}
+            className={className}
+            color={color}
+            block={block}
+            inverted={inverted}
+            disabled={disabled}
+            outlined={outlined}
+            handleClick={handleClick}
+          >
+          {children}
+        </LegacyButton>
+        )
+      }
+      <FontStyle />
+    </div>
+  )
+}
 
-        &:hover:not([disabled]) {
-          background-color: ${theme.colors.bg2};
-        }
-        &:active:not([disabled]) {
-          background-color: ${theme.colors.bg3};
-        }
-      `};
+const Container = styled.button<IButton>(
+  ({disabled, loading, primary, secondary, tertiary, icon, forcedWidth}) => css`
```

Comment:
The formatting of this could be a bit nicer. Prettier should do this for us. Do you have the prettier extension installed in VS Code?

Reply:
Yes I do, will look into this